### ISSUE TITLE
Lita Disconnects from HipCat - Connection Reset by Peer

### ISSUE THREAD

**Issue by MattMencel** [jimmycuadra/lita-hipchat#26]
See https://github.com/jimmycuadra/lita/issues/94#issuecomment-95432567

Lita disconnected from my HipChat room again yesterday morning.

```
D, [2015-05-04T08:02:16.521537 #6426] DEBUG -- : TRYING stanzacbs...
D, [2015-05-04T08:02:16.521568 #6426] DEBUG -- : TRYING message/iq/presence/cbs...
W, [2015-05-04T08:02:21.379626 #6426]  WARN -- : EXCEPTION:
    Errno::ECONNRESET
    Connection reset by peer
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/connection.rb:45:in `sysread'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/connection.rb:45:in `sysread'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/openssl/buffering.rb:61:in `fill_rbuff'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/openssl/buffering.rb:301:in `eof?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/source.rb:235:in `empty?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:148:in `empty?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:196:in `pull_event'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:184:in `pull'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/sax2parser.rb:92:in `parse'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/streamparser.rb:91:in `parse'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/stream.rb:75:in `block in start'
W, [2015-05-04T08:02:21.379789 #6426]  WARN -- : Exception caught in Parser thread! (Errno::ECONNRESET)
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/connection.rb:45:in `sysread'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/connection.rb:45:in `sysread'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/openssl/buffering.rb:61:in `fill_rbuff'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/openssl/buffering.rb:301:in `eof?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/source.rb:235:in `empty?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:148:in `empty?'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:196:in `pull_event'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/baseparser.rb:184:in `pull'
    /opt/rbenv/versions/2.1.5/lib/ruby/2.1.0/rexml/parsers/sax2parser.rb:92:in `parse'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/streamparser.rb:91:in `parse'
    /opt/lita/vendor/ruby/2.1.0/gems/xmpp4r-0.5.6/lib/xmpp4r/stream.rb:75:in `block in start'
```

**Comment by jimmycuadra** [comment]
Probably goes without saying, but just to confirm, this is running lita-hipchat 2.1.2, correct?

**Comment by MattMencel** [comment]
Yes I've got the 2.1.2 gem installed.

**Comment by jimmycuadra** [comment]
I just released lita-hipchat 2.1.3, which will hopefully *actually* shut Lita down so it can be restarted when this happens. Please leave a comment if you see this happen again under 2.1.3! Sorry for the hassle.

**Comment by jimmycuadra** [comment]
Closing due to inactivity – please reopen if you see the crash again.
